tools-local/unix/acquire-tool.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+
+usage()
+{
+    echo "usage: $0 <repository-root> <tool-name> <override-scripts-folder-path>"
+    echo "  repository-root: Path to repository root."
+    echo "  tool-name: Name of the tool to download."
+    echo "  override-scripts-folder-path: If a path is specified then, scripts from the specified folder will be invoked."
+    echo "                                  Otherwise, the default scripts located within the repository will be invoked."
+    echo ""
+    echo "Downloads the declared version of the specified tool from the corresponding URL specified in the .toolversions file."
+    echo "If download succeeds then, returns the path to the executable."
+    echo "Exit 1 if download fails."
+}
+
+if [ $# -ne 3 ]; then
+    usage
+    exit 1
+fi
+
+repoRoot="$1"
+toolName="$2"
+overrideScriptsPath="$3"
+
+if [ -z "$repoRoot" ]; then
+    echo "Argument passed as repository-root is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+if [ -z "$toolName" ]; then
+    echo "Argument passed as tool-name is empty. Please provide a non-empty string."
+    usage
+    exit 1
+fi
+
+scriptPath="$(cd "$(dirname "$0")"; pwd -P)"
+. "$scriptPath/tool-helper.sh"
+declaredVersion="$(get_tool_config_value "$repoRoot" "$toolName" "DeclaredVersion")"
+
+
+# Downloads the package corresponding to the tool, and extracts the package.
+download_extract()
+{
+    # Get the download URL
+    downloadUrl="$(get_tool_config_value "$repoRoot" "$toolName" "DownloadUrl")"
+
+    if [ $? -ne 0 ]; then
+        echo "$downloadUrl"
+        exit 1
+    fi
+
+    downloadPackageFilename=$(get_download_package_name "$repoRoot" "$toolName")
```

Comment:
These get config helper method names should align with the name of the config value they read  e.g. get_download_file